### PR TITLE
Issue 6147: Stream Tag documenation updates.

### LINKED ISSUE #6147: Stream Tag related documentation.
**Describe the solution you'd like**
Update Pravega docs to talk about Stream Tags.

We need to:
Stream Tags are now part of the StreamConfiguration. Ensure the documentation reflects the same.

### PR DESCRIPTION
**Pull request by shrids**
**Change log description**  
Ensure Stream Tag option, which is now a part of the StreamConfiguration is updated as part of the Controller documentation.

**Purpose of the change**  
Fixes #6147 

**What the code does**  
Ensure Stream Tag option, which is now a part of the StreamConfiguration is updated as part of the Controller documentation
Note: REST API related documentation is auto-generated and will be updated by PR #6143 

**How to verify it**  
No changes to functionality.

### CODE REVIEW

**Review by shiveshr — APPROVED** [review]

File: documentation/src/docs/controller-service.md [hunk 1]
```diff
@@ -128,7 +128,11 @@ have two policies that users can define, namely [**Scaling** **Policy**](https:/
        - **Scaling policy** describes if and under what circumstances a Stream should automatically scale its number of Segments.  
        - **Retention policy** describes a policy about how much data to retain within a Stream based on **time** (*Time-based Retention*) and data **size** (*Size-based Retention*).
 
-  3. [**Transaction**](pravega-concepts.md#transactions) **Management**: Implementing Transactions requires the manipulation of Stream Segments. With
+  3. **Tag Management**: The Controller is responsible for storing user-defined Tags for a given Stream.
+     Tags are singleton labels that are useful to specify identifying attributes of a Stream, meaningful and relevant to users.
+     The Controller indexes these Tags and provides APIs by which the user  can organize and select a subset of Streams under a given Scope.
```
Comment (shiveshr) [inline documentation/src/docs/controller-service.md:133]: perhaps we need to say organize and select a subset of streams under a scope that share the same tag.

Comment (shrids) [reply documentation/src/docs/controller-service.md]: done.

File: documentation/src/docs/controller-service.md [hunk 2]
```diff
@@ -634,7 +639,7 @@ and `updateStream()` operation is performed.
 - Once the update Stream processing starts, it first sets the Stream state to *Updating*.
 - Then, the Stream configuration is updated in the metadata store followed by notifying
 Segment Stores for all _active_ Stream Segments of the Stream, about the change in
-policy. Now the state is reset to *Active*.
+policy. Any Tag related updates causes the Tag reverse indexes to be updated with the newer Tags. Now the state is reset to *Active*.
```
Comment (shiveshr) [inline documentation/src/docs/controller-service.md:642]: we have mentioned indexes earlier and reverse index here. we should perhaps describe how we index it and call it reverse index consistently.

Comment (shrids) [reply documentation/src/docs/controller-service.md]: improved it.

**Pull request merged**